packages/explorer/src/app/(explorer)/[chainName]/worlds/WorldsForm.tsx

```diff
@@ -0,0 +1,136 @@
+"use client";
+
+import Image from "next/image";
+import { useParams, useRouter } from "next/navigation";
+import { Address, isAddress } from "viem";
+import * as z from "zod";
+import { Command as CommandPrimitive } from "cmdk";
+import { useState } from "react";
+import { useForm } from "react-hook-form";
+import { zodResolver } from "@hookform/resolvers/zod";
+import { Button } from "../../../../components/ui/Button";
+import { Command, CommandGroup, CommandItem, CommandList } from "../../../../components/ui/Command";
+import { Form, FormControl, FormField, FormItem, FormMessage } from "../../../../components/ui/Form";
+import { Input } from "../../../../components/ui/Input";
+import mudLogo from "../../icon.svg";
+import { getWorldUrl } from "../../utils/getWorldUrl";
+
+const formSchema = z.object({
+  worldAddress: z
+    .string()
+    .refine((value) => isAddress(value), {
+      message: "Invalid world address",
+    })
+    .transform((value): Address => value as Address),
+});
+
+export function WorldsForm({ worlds }: { worlds: Address[] }) {
+  const router = useRouter();
+  const { chainName } = useParams();
+  const [open, setOpen] = useState(false);
+
+  const form = useForm<z.infer<typeof formSchema>>({
+    resolver: zodResolver(formSchema),
+    reValidateMode: "onChange",
+  });
+
+  function onSubmit({ worldAddress }: z.infer<typeof formSchema>) {
+    router.push(getWorldUrl(chainName as string, worldAddress));
+  }
+
+  function onLuckyWorld() {
+    if (worlds.length > 0) {
+      const luckyAddress = worlds[Math.floor(Math.random() * worlds.length)];
+      router.push(getWorldUrl(chainName as string, luckyAddress));
+    }
+  }
+
+  return (
+    <div className="mx-auto flex min-h-screen w-[450px] flex-col items-center justify-center p-4">
+      <h1 className="mb-6 flex items-center gap-4 self-start font-mono text-4xl font-bold uppercase">
+        <Image src={mudLogo} alt="MUD logo" width={48} height={48} /> Worlds Explorer
+      </h1>
+
+      <Command className="overflow-visible">
+        <Form {...form}>
+          <form onSubmit={form.handleSubmit(onSubmit)} className="w-full max-w-md space-y-6">
```

Comment:
ooc why do we set a max width here and also a hardcoded width above?

Reply:
good catch, forgot to remove